python/orca/example/learn/mmcv/faster_rcnn/train.py

```diff
@@ -0,0 +1,299 @@
+import os
+
+import numpy as np
+import os.path as osp
+
+from bigdl.orca import init_orca_context, stop_orca_context
+from bigdl.orca.data.file import get_remote_dir_to_local
+from bigdl.orca.learn.pytorch.experimential.mmcv.mmcv_ray_estimator import MMCVRayEstimator
+
+import mmcv
+from mmcv import Config
+
+from mmdet.apis import set_random_seed
+from mmdet.datasets import build_dataloader, build_dataset, replace_ImageToTensor
+
+
+def mmcv_runner_creator(cfg):
+    from mmdet.datasets.builder import DATASETS
+    from mmdet.datasets.custom import CustomDataset
+    from mmdet.models import build_detector
+    from mmdet.utils import (compat_cfg, find_latest_checkpoint, get_root_logger)
+    from mmdet.core import DistEvalHook, EvalHook, build_optimizer
+    from mmcv.runner import (DistSamplerSeedHook, EpochBasedRunner,
+                             Fp16OptimizerHook, OptimizerHook, build_runner)
+
+    @DATASETS.register_module()
+    class KittiTinyDataset(CustomDataset):
+        CLASSES = ('Car', 'Pedestrian', 'Cyclist')
+
+        def load_annotations(self, ann_file):
+            cat2label = {k: i for i, k in enumerate(self.CLASSES)}
+            # load image list from file
+            image_list = mmcv.list_from_file(self.ann_file)
+
+            data_infos = []
+            # convert annotations to middle format
+            for image_id in image_list:
+                filename = f'{self.img_prefix}/{image_id}.jpeg'
+                image = mmcv.imread(filename)
+                height, width = image.shape[:2]
+
+                data_info = dict(filename=f'{image_id}.jpeg', width=width, height=height)
+
+                # load annotations
+                label_prefix = self.img_prefix.replace('image_2', 'label_2')
+                lines = mmcv.list_from_file(osp.join(label_prefix, f'{image_id}.txt'))
+
+                content = [line.strip().split(' ') for line in lines]
+                bbox_names = [x[0] for x in content]
+                bboxes = [[float(info) for info in x[4:8]] for x in content]
+
+                gt_bboxes = []
+                gt_labels = []
+                gt_bboxes_ignore = []
+                gt_labels_ignore = []
+
+                # filter 'DontCare'
+                for bbox_name, bbox in zip(bbox_names, bboxes):
+                    if bbox_name in cat2label:
+                        gt_labels.append(cat2label[bbox_name])
+                        gt_bboxes.append(bbox)
+                    else:
+                        gt_labels_ignore.append(-1)
+                        gt_bboxes_ignore.append(bbox)
+
+                data_anno = dict(
+                    bboxes=np.array(gt_bboxes, dtype=np.float32).reshape(-1, 4),
+                    labels=np.array(gt_labels, dtype=np.long),
+                    bboxes_ignore=np.array(gt_bboxes_ignore,
+                                           dtype=np.float32).reshape(-1, 4),
+                    labels_ignore=np.array(gt_labels_ignore, dtype=np.long))
```

Comment:
This is removed after numpy 1.24, just use `int`

Referene: https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated